src/nfa.rs

```diff
@@ -322,6 +323,42 @@ impl<S: StateID> State<S> {
     }
 }
 
+#[derive(Clone, Debug)]
+struct Dense<S>(Vec<S>);
+
+impl<S> Dense<S>
+    where S: StateID,
```

Comment:
I prefer keeping the style consistent with the surround code. Could you write `impl<S: StateID> Dense<S>` instead? (I generally only use `where` clauses when the bounds get more complicated.)

Reply:
Fixed